docs/modules/clusters/pages/user-code-namespaces.adoc

```diff
@@ -0,0 +1,36 @@
+= {ucn}
+:description: {ucn} provide a container for Java classpath resources, such as user code and accompanying artifacts like property files. This provides namespace isolation to ensure that access to resources in different namespaces can be managed through configuration.
+:page-enterprise: true
+:page-beta: true
+
+{description}
+
+NOTE: This approach replaces the deprecated {ucd} method of adding your resources to make them available for your user-defined functions. For information on migrating your {ucd} to {ucn}, see the xref:clusters:ucn-migrate-ucd.adoc[] topic. The original {ucd} information has been retained to support your migration in the xref:clusters:legacy-ucd.adoc[] section. 
+
+Your resources can be run on a Hazelcast cluster to support your business logic. For example, you might want to use your resources in one of the following:
+
+* MapStore
+* Split brain events
+* `ExecutorService` runnable
+
+To use {ucn}, you must enable it as described in the xref:clusters:ucn-enable.adoc[] topic.
+
+User-defined functions are associated with {ucn} using the namespace names. Named data structures each define a single namespace name, or you can share a namespace by using a wildcard in the configuration. You can also configure the `default` namespace; for example, to provide resources for a user-defined function that is not associated with a data structure. For further information on the `default` namespace, see the xref:clusters:ucn-non-associated.adoc[] topic. Your resources are located using an URL. For example, your resources could be located on your member or client with an URL such as _\file:///opt/hz/lib/calc.jar_ or _classpath:///com/acme/myClass_. 
```

Comment:
> you can share a namespace by using a wildcard in the configuration

I (genuinely) didn't know it did this

Reply:
Namespaces do not support wildcards - each data structure needs to define an explicit Namespace to associate with, otherwise it will associate with the `default` Namespace. A single Namespace (such as `ns1`) can be used with any number of data structures, as it's just a pointer towards class resources and contains nothing data structure specific in itself.